reflex/experimental/layout.py

```diff
@@ -0,0 +1,212 @@
+
+    open: bool = True
+
+    def toggle(self):
+        """Toggle the sidebar."""
+        self.open = not self.open
+
+    @classmethod
+    def get_component(cls, *children, **props):
+        """Get the stateful sidebar component.
+
+        Args:
+            children: The children components.
+            props: The properties of the sidebar.
+
+        Returns:
+            The stateful sidebar component.
+        """
+        return Sidebar.create(*children, **props)
+
+
+class DrawerSidebar(DrawerRoot):
+    """A component that renders a drawer sidebar."""
+
+    @classmethod
+    def create(cls, *children, **props):
+        """Create the sidebar component.
+
+        Args:
+            children: The children components.
+            props: The properties of the sidebar.
+
+        Returns:
+            The drawer sidebar component.
+        """
+        direction = props.pop("direction", "left")
+        props.setdefault("border_right", f"1px solid {color('accent', 12)}")
+        props.setdefault("background_color", color("accent", 1))
+        props.setdefault("width", "20vw")
+        props.setdefault("height", "100vh")
+        return super().create(
+            drawer.trigger(
+                IconButton.create(
+                    "arrow-right-from-line",
+                    background_color="transparent",
+                ),
+                position="absolute",
+                top="15",
+                left="15",
+            ),
+            drawer.portal(
+                drawer.content(
+                    *children,
+                    **props,
+                )
+            ),
+            direction=direction,
+        )
+
+
+class SidebarTrigger(Fragment):
+    """A component that renders the sidebar trigger."""
+
+    @classmethod
+    def create(cls, sidebar: Component, **props):
+        """Create the sidebar trigger component.
+
+        Args:
+            sidebar: The sidebar component.
+            props: The properties of the sidebar trigger.
+
+        Returns:
+            The sidebar trigger component.
+        """
+        trigger_props = {
+            **props,
+            "position": "absolute",
+            "z_index": "15",
+            "top": "15",
+            "background_color": "transparent",
+        }
+
+        if sidebar.State:
+            open, toggle = sidebar.State.open, sidebar.State.toggle  # type: ignore
+        else:
+            open, toggle = Var.create("open"), call_script(Var.create("setOpen(!open)"))  # type: ignore
+        trigger_props["left"] = cond(open, "calc(20vw - 32px)", "15")
+
+        trigger = cond(
+            open,
+            IconButton.create(
+                "arrow-left-from-line",
+                on_click=toggle,
+                **trigger_props,
+            ),
+            IconButton.create(
+                "arrow-right-from-line",
+                on_click=toggle,
+                **trigger_props,
+            ),
```

Comment:
these don't show up in light mode, for whatever reason. probably need the color to be based on the theme gray color with alpha.